server/tilesource/__init__.py

```diff
@@ -20,21 +20,42 @@
 # Deal with a bug where PEP257 crashes when parsing __all__
 # flake8: noqa
 
+import collections
+from girder.constants import TerminalColor
+from girder import logger
+
 from .base import TileSource, TileSourceException, \
     TileSourceAssetstoreException
 
-try:
-    from .test import TestTileSource
-except ImportError:
-    from .dummy import DummyTileSource as TestTileSource
-try:
-    from .tiff import TiffGirderTileSource
-except ImportError:
-    from .dummy import DummyTileSource as TiffGirderTileSource
-try:
-    from .svs import SVSGirderTileSource
-except ImportError:
-    from .dummy import DummyTileSource as SVSGirderTileSource
+AvailableTileSources = collections.OrderedDict()
+all = [TileSource, TileSourceException, TileSourceAssetstoreException,
+       AvailableTileSources]
+
+sourceList = [
+    {'moduleName': '.tiff', 'className': 'TiffGirderTileSource'},
```

Comment:
This list is our python module name and has nothing to do with extensions.

I agree, relying on file extensions is a bad idea.  We currently work our way through the readers, asking each in turn if they can handle the file.  We could use mime type (or even file extension) to change the order of the trials, but I think the current method is sound enough.